### ISSUE TITLE
Missing range check for the parameter eps.

### ISSUE THREAD

**Issue by Zoeeeeey** [pytorch/pytorch#123525]
### Description
In many APIs, there exists a parameter called `eps`, which is designed to be a small floating-point number. However, I have found that the code still runs when given a large integer value. 

I believe this is a bug, as it contradicts the original purpose of `eps`.

Furthermore, when the value of `eps` is very large, it artificially inflates the denominator in some normalization algorithms, resulting in normalization outcomes that are very close to zero. 

This might cause issues such as gradient vanishing, slower convergence, or poor model performance in practical applications.

### Poc
Sample code to reproduce the problem:
```Python
input = torch.randn(5, 10)
m = torch.nn.InstanceNorm1d(num_features=5, eps=2018915490, affine=True)
output = m(input)
print(output)
```

The error message i got:
```
tensor([[ 1.8597e-06, -1.0711e-05, -1.5410e-05, -1.1158e-05,  5.3511e-05,
         -4.6150e-05, -8.2182e-06,  1.1602e-05, -1.1196e-05,  3.5870e-05],
        [-2.5561e-05, -5.4325e-06, -3.1732e-07,  3.3941e-05, -1.5833e-05,
         -1.4249e-05, -6.0614e-06,  2.2347e-05, -3.4462e-05,  4.5628e-05],
        [ 2.5948e-06,  9.5551e-06,  1.5357e-06,  2.2961e-05,  1.8517e-05,
          4.8422e-06, -2.6347e-06, -3.4661e-05, -1.5167e-05, -7.5434e-06],
        [ 4.2461e-06, -2.4960e-05,  9.2949e-06, -2.0175e-05, -4.8853e-06,
         -1.5256e-05,  2.9005e-05,  7.3481e-06,  9.9568e-06,  5.4246e-06],
        [-2.1944e-05,  1.2490e-06, -3.5701e-06, -7.2909e-06,  6.0630e-07,
         -2.8139e-05, -7.7213e-06,  6.3160e-06,  3.5687e-05,  2.4807e-05]],
       grad_fn=<SqueezeBackward1>)
```

### Supplement
I initially observed this phenomenon with the `torch.nn.InstanceNorm1d`. And, upon verification, I found that similar occurrences are present in many other APIs.
Including:
```
torch.nn.InstanceNorm1d, torch.nn.InstanceNorm2d, torch.nn.InstanceNorm3d,
torch.nn.LazyInstanceNorm1d, torch.nn.LazyInstanceNorm2d, torch.nn.LazyInstanceNorm3d,
torch.nn.GroupNorm, torch.nn.PairwiseDistance,
torch.nn.BatchNorm1d, torch.nn.BatchNorm2d, torch.nn.BatchNorm3d,
torch.nn.LazyBatchNorm1d, torch.nn.LazyBatchNorm2d, torch.nn.LazyBatchNorm3d,
torch.nn.LayerNorm, torch.nn.functional.normalize, torch.nn.functional.gaussian_nll_loss
torch.nn.GaussianNLLLoss, torch.nn.utils.spectral_norm
```

Sample code to repoduce the problem with `torch.nn.utils.spectral_norm`:
```Python
import torch
import torch.nn as nn
import torch.nn.utils as nn_utils

input = torch.randn(3, 10)
linear = nn.Linear(in_features=10, out_features=5)
linear = nn_utils.spectral_norm(linear, eps=2018915490)
output = linear(input)
print(output)
```

The error message i got:
```
tensor([[nan, nan, nan, nan, nan],
        [nan, nan, nan, nan, nan],
        [nan, nan, nan, nan, nan]], grad_fn=<AddmmBackward0>)
```

### Expectation
I suggest implementing a range check for the `eps` parameter to avoid the inadvertent passing of large integers without a warning, leading to suboptimal results.

### Versions

Collecting environment information...
PyTorch version: 2.2.2+cu121
Is debug build: False
CUDA used to build PyTorch: 12.1
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.5 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Clang version: Could not collect
CMake version: version 3.16.3
Libc version: glibc-2.31

Python version: 3.9.19 (main, Mar 21 2024, 17:11:28)  [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-5.4.0-169-generic-x86_64-with-glibc2.31
Is CUDA available: True
CUDA runtime version: 12.1.66
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 2080 Ti
Nvidia driver version: 535.146.02
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.6.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.6.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.6.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.6.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.6.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.6.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.6.0
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                       x86_64
CPU op-mode(s):                     32-bit, 64-bit
Byte Order:                         Little Endian
Address sizes:                      46 bits physical, 48 bits virtual
CPU(s):                             96
On-line CPU(s) list:                0-95
Thread(s) per core:                 2
Core(s) per socket:                 24
Socket(s):                          2
NUMA node(s):                       2
Vendor ID:                          GenuineIntel
CPU family:                         6
Model:                              85
Model name:                         Intel(R) Xeon(R) Platinum 8255C CPU @ 2.50GHz
Stepping:                           7
Frequency boost:                    enabled
CPU MHz:                            3099.999
CPU max MHz:                        2501.0000
CPU min MHz:                        1000.0000
BogoMIPS:                           5000.00
Virtualization:                     VT-x
L1d cache:                          1.5 MiB
L1i cache:                          1.5 MiB
L2 cache:                           48 MiB
L3 cache:                           71.5 MiB
NUMA node0 CPU(s):                  0-23,48-71
NUMA node1 CPU(s):                  24-47,72-95
Vulnerability Gather data sampling: Mitigation; Microcode
Vulnerability Itlb multihit:        KVM: Mitigation: Split huge pages
Vulnerability L1tf:                 Not affected
Vulnerability Mds:                  Not affected
Vulnerability Meltdown:             Not affected
Vulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable
Vulnerability Retbleed:             Mitigation; Enhanced IBRS
Vulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence
Vulnerability Srbds:                Not affected
Vulnerability Tsx async abort:      Mitigation; TSX disabled
Flags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cdp_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm mpx rdt_a avx512f avx512dq rdseed adx smap clflushopt clwb intel_pt avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local dtherm ida arat pln pts pku ospke avx512_vnni md_clear flush_l1d arch_capabilities

Versions of relevant libraries:
[pip3] numpy==1.26.4
[pip3] torch==2.2.2
[pip3] torchaudio==2.2.2
[pip3] torchvision==0.17.2
[pip3] triton==2.2.0
[conda] numpy                     1.26.4                   pypi_0    pypi
[conda] torch                     2.2.2                    pypi_0    pypi
[conda] torchaudio                2.2.2                    pypi_0    pypi
[conda] torchvision               0.17.2                   pypi_0    pypi
[conda] triton                    2.2.0                    pypi_0    pypi

**Comment by Zoeeeeey** [comment]
These are some test cases.
https://gist.github.com/Zoeeeeey/386d562728a735311c36de359dbfdc1d

**Comment by qqaatw** [comment]
I do not think this is a bug as eps is the discretion of users, and runtime range check can impose overheads.
